src/xdp/program.c

```diff
@@ -86,6 +92,8 @@ typedef struct _XDP_PROGRAM_FRAME_CACHE {
         IPV6_HEADER *Ip6Hdr;
     };
     UDP_HDR *UdpHdr;
+    TCP_HDR *TcpHdr;
+    UINT8 *TcpHdrOptions;
```

Comment:
Can we assume (and validate) the header options are contiguous with the header? That would probably simplify things.

Reply:
If we can assume the entire TCP header and options are contiguous, we would simplify a few things. But, only assuming header options are contiguous doesn't seem like it would simplify much. Wouldn't we still need to call into `XdpParseFragmentedTcp` and fetch the first 20 bytes to see how much more header space TCP takes and then fetch what's beyond the 20 bytes?

Reply:
Right, if we call that function twice, we'd either need to keep these fields split, potentially re-copy fragmented data twice, or add complex optimizations to partially copy into the storage buffer. If we can just remove the options handling entirely for now, that seems easiest.